webapp/channels/src/utils/markdown/renderer.tsx

```diff
@@ -40,23 +40,29 @@ export default class Renderer extends marked.Renderer {
         let searchedContent = '';
 
         if (this.formattingOptions.searchPatterns) {
-            const tokens = new Map();
-
-            let searched = TextFormatting.sanitizeHtml(code);
-            searched = TextFormatting.highlightSearchTerms(
-                searched,
-                tokens,
-                this.formattingOptions.searchPatterns,
-            );
-
-            if (tokens.size > 0) {
-                searched = TextFormatting.replaceTokens(searched, tokens);
-
-                searchedContent = (
-                    '<div class="post-code__search-highlighting">' +
-                        searched +
-                    '</div>'
+            try {
```

Comment:
Nit: Can this whole block potentially fail in a way where we'd want to catch it? Usually I like to limit `try` blocks to just the call that we want to catch.

Reply:
Potentially yes. That's why the catch has code to verify the error is the one we are trying to catch and if not, throw it again.

Regarding only covering what may throw the error, I feel it adds more boilerplate, and make it less readable. If we only wrap `highlightSearchTerm`, we need to handle that tokens may have elements, but we don't want to add that to the `searchedContent`. So we have to make more complicated the `if (tokens.size > 0)`. If we include the if inside the try block, the first two lines (44 and 46) look out of place, specially because they are variable definitions only used inside the try block. So finally, I thought it was more readable to have the whole thing in the same place.